docs/programming/tilebuffer.md

```diff
@@ -0,0 +1,285 @@
+
+Don't forget to export the [palette](./palette.md) as `overworld.gpl`
+
+The palette file is ready for use here : [overworld.gpl](./res/overworld.gpl)
+
+### Prepare the map
+
+You need to use [Tiled](https://www.mapeditor.org/)
+
+And install this addon : https://github.com/youenchene/tiled-export-to-amiga
+
+1. Go to `Settings`
+2. Then `Plugins`
+3. Click Open after the Directory, and copy inside `gogitum.mjs`  
+
+On **Tiled** :
+
+1. Create new Project
+2. Click on `New Tileset...`
+3. Load `overworld.png` set it to 16 x 16
+4. Then `New Map...`
+5. Choose `Orthogonal`, `CSV` and `Right Down`, `200` tiles x `100` tiles and `16px` tile size
+6. Draw a map
+7. `File > Export As...`
+8. Choose `GOGITUM Export ACE - pTileData (UBYTE)` format and export as `overworld.dat` in your res `folder`.
+
+
+The map file is ready for use here : [overworld.dat](./res/overworld.dat)
+
+### Recap
+
+In your `res` folder you should have :
+
+- `overworld.dat` (map)
+- `overworld.gpl` (palette)
+- `overworld.png` (tiles)
+
+## Let's Code
+
+### Assets Preprocessing
+
+First, append your `CMakeLists.txt` file and add at the end. 
+
+It loads and converts the palette and then use it to convert your tiles file. It also loads the map file.
+
+```cmake
+# Convert palette and background image
+convertPalette(${GAME_LINKED} ${RES_DIR}/overworld.gpl ${DATA_DIR}/overworld.plt)
+convertBitmaps(
+  TARGET ${GAME_LINKED} PALETTE ${RES_DIR}/overworld.gpl
+  INTERLEAVED SOURCES ${RES_DIR}/overworld.png
+  DESTINATIONS ${DATA_DIR}/overworld.bm
+)
+
+file(COPY ${RES_DIR}/overworld.dat DESTINATION ${DATA_DIR})
+```
+
+### Update your game.c
+
+Then on on your `game.c` or equivalent.
+
+Declare constants and main static variable
+
+```c
+#define TILE_MAP_SIZE_X 200
+#define TILE_MAP_SIZE_Y 80
+#define GAME_BPP 5 // 32 colors
+#define SCORE_HEIGHT 16
+
+
+// View containing all the viewports
+static tView *s_pView; 
+// For score - won't be used in this tuto
+static tVPort *s_pVpScore; // Viewport for score
+static tSimpleBufferManager *s_pScoreBuffer;
+// Viewport for playfield
+static tVPort *s_pVpMain; 
+// Your buffer manager, it will handle your scrolling
+static tTileBufferManager *s_pMainBuffer;
+// Tiles
+static tBitMap *s_pTiles;
+// Palette
+static UWORD s_pPalette[1 << GAME_BPP];
+
+static UWORD s_uwMapTileWidth;
+static UWORD s_uwMapTileHeight;
+//[...]
+// To controle camera/view on scrolling
+UWORD g_uCameraX=0;
+UWORD g_uCameraY=0;
+UWORD g_uCameraSpeed=2;
+//[...]
+
+```
+
+Then add a method to load your map. The map is loaded column by column for speed loading speed purpose :
+
+```c
+static void loadMap(void) {
+	logWrite("Map Loading...\n");
+	tFile *pFileTilemap = diskFileOpen("data/overworld.dat", "rb");
+	fileRead(pFileTilemap, &s_uwMapTileWidth, sizeof(s_uwMapTileWidth));
+	fileRead(pFileTilemap, &s_uwMapTileHeight, sizeof(s_uwMapTileHeight));
+	s_uwMapTileWidth++;
+	s_uwMapTileHeight++;
```

Comment:
From what I understand, you are the author of tiled export plugin (congrats!). Perhaps it would be better to change the plugin code so that it has those values already bigger by one on the file?

Reply:
https://github.com/youenchene/tiled-export-to-amiga/issues/1